src/lib/assets/hooks/collectibles.ts

```diff
@@ -15,218 +13,111 @@ import { getAccountAssetsStoreKey } from 'app/store/tezos/assets/utils';
-  const tezCollectibles = useAllTezosAccountCollectiblesSlugs(accountTezAddress);
-  const evmCollectibles = useAllEvmAccountCollectiblesSlugs(accountEvmAddress);
+      accountCollectibles = accountCollectibles.concat(
+        Object.entries(storedRaw).map<AccountCollectible>(([slug, { status }]) => ({
+          slug,
+          status: getAssetStatus(balances[slug], status),
+          chainId
+        }))
+      );
+    }
 
-  return useMemo(() => [...tezCollectibles, ...evmCollectibles], [tezCollectibles, evmCollectibles]);
+    return accountCollectibles.filter(c => c.status !== 'removed');
+  }, [enabledChains, account, storedRecord, balancesRecord]);
 };
 
-const useTezosChainAccountCollectibles = (account: string, chainId: string) => {
+/**
+ * @returns Collectibes of account for chain, that were not removed
+ */
+export const useTezosChainAccountCollectibles = (account: string, chainId: string) => {
   const stored = useChainAccountCollectiblesSelector(account, chainId);
 
   const balances = useAllAccountBalancesSelector(account, chainId);
 
-  return useMemoWithCompare<AccountCollectible[]>(
-    () => {
-      const result: AccountCollectible[] = [];
-
-      for (const [slug, { status }] of Object.entries(stored)) {
-        if (status !== 'removed') result.push({ slug, status: getAssetStatus(balances[slug], status), chainId });
-      }
-
-      return result;
-    },
-    [stored, balances, chainId],
-    isEqual
-  );
-};
-
-export const useEnabledTezosAccountCollectiblesSlugs = (publicKeyHash: string) => {
-  const collectibles = useTezosAccountCollectibles(publicKeyHash);
-
-  return useMemo(
-    () =>
-      collectibles
-        .filter(({ status }) => status === 'enabled')
-        .map(({ chainId, slug }) => toChainAssetSlug(TempleChainKind.Tezos, chainId, slug)),
-    [collectibles]
-  );
-};
-
-export const useAllTezosAccountCollectiblesSlugs = (publicKeyHash: string) => {
-  const collectibles = useTezosAccountCollectibles(publicKeyHash);
-
-  return useMemo(
-    () =>
-      collectibles
-        .filter(({ status }) => status !== 'removed')
-        .map(({ chainId, slug }) => toChainAssetSlug(TempleChainKind.Tezos, chainId, slug)),
-    [collectibles]
-  );
-};
-
-export const useEnabledTezosChainAccountCollectiblesSlugs = (publicKeyHash: string, chainId: string) => {
-  const collectibles = useTezosChainAccountCollectibles(publicKeyHash, chainId);
+  return useMemo<AccountCollectible[]>(() => {
+    const result: AccountCollectible[] = [];
 
-  return useMemo(
-    () => collectibles.filter(({ status }) => status === 'enabled').map(({ slug }) => slug),
-    [collectibles]
-  );
-};
-
-export const useAllTezosChainAccountCollectiblesSlugs = (publicKeyHash: string, chainId: string) => {
-  const collectibles = useTezosChainAccountCollectibles(publicKeyHash, chainId);
+    for (const [slug, { status }] of Object.entries(stored)) {
```

Comment:
Should we use `Object.entries(stored).filter(...).map(...)` here instead for better performance?